Iops 1514 profiling nhs digital capability statement

KazeemHamzat opened the pull request:
1. Changed the canonical URL for all profiles stated in CapabilityStatement
2. Removed the NHSDigital-Identifier profile to conform to IOPS Policy of no profiling for Identifier.
3. Removed the NHSDigital-Reference profile. Reference.Identifier is the only differentia on it, and mapped to NHSDigital-Identifier. 
4. Changed all profiles that referenced NHSDigital-Reference profile to HL7 Reference base profile.
5. Changed the canonical URL for all ValueSets.
6. All ValueSets and Examples are now validating.

KazeemHamzat committed:
Updating Canonical URL of CapabilityStatement

Updating Canonical URL of CapabilityStatement to England

KazeemHamzat committed:
Update fhirpkg.lock.json

Updating the package manifest to 0.0.3 pre-release

KazeemHamzat committed:
Update Extension-NHSDigital-CapabilityStatement-Package.xml

Updating Canonical URL to England in Packages for Extension and Examples

KazeemHamzat committed:
Updating affected Target Profile

Updating affected Target Profiles

KazeemHamzat committed:
Further update of some Canonical URL

Further update of Canonical URL:
Location, ServiceRequest, Practitioner & PractitionerRole

KazeemHamzat committed:
Further update of CURL

Further update of CURL

KazeemHamzat committed:
Update of CURL

Update of CURL:
Observation, NHSDigital-PractitionerRole-Minimal and ValueSets: England-CareSettingType and /NHSDigital-message-event

KazeemHamzat committed:
Update of CURL

Checking that Canonical URL of all England SD in Capability Statement have been updated.

KazeemHamzat committed:
Update of CURL

Update of CURL in MessageDefinition Folder

KazeemHamzat committed:
Updating CURL

Updating CURL for DocumentReference Profile

KazeemHamzat committed:
Update CURL MedicarionRequestCodeaable Concept

MedicarionRequestCodeaable Concept,Dosage,ValueSets

KazeemHamzat committed:
Update of CURL MessageDefinition.event

MessageDefinition.event

KazeemHamzat committed:
Updating CURL for Valusets

England-MedicationRequest-Code, England-usage-context-type

KazeemHamzat committed:
Update SD-England-Reference

Removing reference to Identifier profile.

KazeemHamzat committed:
Delete NHSDigital-Reference.StructureDefinition.xml

Removing SD for Reference Profile

KazeemHamzat committed:
Updating CURL: Extensions

Updating CURL: Extensions

KazeemHamzat committed:
Update pointing Profile Ref to HL7

Update pointing Profile Ref to HL7

KazeemHamzat committed:
Updating CURL for Observation & Composition

Updating CURL for Observation & Composition profiles

KazeemHamzat committed:
Update CURL for Profiles

Update CURL for Profiles

KazeemHamzat committed:
Update CURL for Task

Update CURL for Task

KazeemHamzat committed:
Update to package Manifest

KazeemHamzat committed:
Update fhirpkg.lock.json

package manifest

declankieran-nhsd commented on CapabilityStatement/CapabilityStatement-nhsdigital.json:
```diff
@@ -1,13 +1,14 @@
     "resourceType": "CapabilityStatement",
-    "id": "nhsdigital-requirements",
-    "url": "https://fhir.nhs.uk/CapabilityStatement/nhsdigital-requirements",
+    "id": "England-CapabilityStatement-IGrequirements",
```
Is the case correct for this? England-CapabilityStatement-IGrequirements, should it be England-CapabilityStatement-IGRequirements?  Also is IGRequirements a useful part of the name here.  Would NHSEngland-CapabilityStatement as an id and name not make more sense?  Same for instances below

declankieran-nhsd commented on CapabilityStatement/CapabilityStatement-nhsdigital.json:
```diff
@@ -33,59 +34,59 @@
                 {
                     "type": "Account",
-                    "profile": "https://fhir.nhs.uk/StructureDefinition/NHSDigital-Account"
+                    "profile": "https://fhir.nhs.uk/StructureDefinition/England-Account"
```
Not NHSEngland?  England seems a bit broad...  Same for all following instances

KazeemHamzat replied:
DA decision was: https://fhir.nhs.uk/ResourceType/England-NAME

declankieran-nhsd commented on StructureDefinition/NHSDigital-MedicationRequest.StructureDefinition.xml:
```diff
@@ -387,14 +387,14 @@
       <type>
         <code value="Reference" />
-        <profile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Reference" />
+        <profile value="http://hl7.org/fhir/StructureDefinition/Reference" />
```
Why is this profiled to point to reference as opposed to a reference to some resource?  Does this even need to be here if no profile is needed?

KazeemHamzat replied:
DA decision was: https://fhir.nhs.uk/ResourceType/England-NAME

declankieran-nhsd commented on StructureDefinition/NHSDigital-Observation.StructureDefinition.xml:
```diff
@@ -152,11 +152,11 @@
         <code value="Reference" />
-        <profile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Reference" />
-        <targetProfile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Organization" />
+        <profile value="http://hl7.org/fhir/StructureDefinition/Reference" />
```
Again, not sure what this is here for?

KazeemHamzat replied:
Removed http://hl7.org/fhir/StructureDefinition/Reference

KazeemHamzat committed:
Resolving the PR Comments

Resolving the PR Comments to remove HL7 Reference profiles: http://hl7.org/fhir/StructureDefinition/Reference

omarisgreat commented on StructureDefinition/NHSDigital-Identifier.StructureDefinition.xml:
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <StructureDefinition xmlns="http://hl7.org/fhir">
-  <url value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Identifier" />
```
Shouldn't this be removed as we are no longer Profiling Identifier?

omarisgreat commented on StructureDefinition/NHSDigital-Immunization.StructureDefinition.xml:
```diff
@@ -227,95 +225,76 @@
-        <targetProfile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Organization" />
-        <targetProfile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-PractitionerRole" />
-        <targetProfile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Practitioner" />
+        <profile value="http://hl7.org/fhir/StructureDefinition/Reference" />
```
Should remove profile tag as this will default to the HL7 Reference anyway

omarisgreat commented on StructureDefinition/NHSDigital-Immunization.StructureDefinition.xml:
```diff
@@ -227,95 +225,76 @@
       </constraint>
       <mustSupport value="true" />
     </element>
-    <element id="Immunization.performer.actor.type">
```
Removal of these constraints seems to go outside the scope of canonical URL renaming, was this removed because it was redundant or was there another reason?

omarisgreat commented on StructureDefinition/NHSDigital-Location.StructureDefinition.xml:
```diff
@@ -1,6 +1,6 @@
 <StructureDefinition xmlns="http://hl7.org/fhir">
-  <url value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Location" />
+  <url value="https://fhir.nhs.uk/StructureDefinition/England-Location" />
   <name value="NHSDigitalLocation" />
```
Should be EnglandLocation

omarisgreat commented on StructureDefinition/NHSDigital-PractitionerRole-Minimal.StructureDefinition.xml:
```diff
@@ -1,15 +1,15 @@
   <id value="51533f7f-1fa3-4866-a200-01618d377455" />
-  <url value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-PractitionerRole-Minimal" />
+  <url value="https://fhir.nhs.uk/StructureDefinition/England-PractitionerRole-Minimal" />
   <name value="NHSDigitalPractitionerRoleMinimal" />
```
All structure definitions with NHSDigital in the name should be updated, I won't be adding comments to the rest for the same issue

omarisgreat commented on StructureDefinition/NHSDigital-PractitionerRole.StructureDefinition.xml:
```diff
@@ -85,7 +85,7 @@
       <definition value="This **MUST** either be a reference to a Practitioner resource or an identifier reference plus a display. In both cases the referenced resource or identfier reference should contain a professional code. &#xD;&#xA;&#xD;&#xA;It is recommended an *identifier reference* and *display* (practioner full name) is included. This identifier should be the Practitioner's primary prof ...
```
Definition uses incorrect name for profile

omarisgreat commented on StructureDefinition/NHSDigital-ServiceRequest.StructureDefinition.json:
```diff
@@ -461,7 +461,7 @@
             "code": "Reference",
             "targetProfile": [
               "http://hl7.org/fhir/StructureDefinition/Resource",
-              "https://fhir.nhs.uk/StructureDefinition/NHSDigital-DocumentReference"
```
DocumentReference target seems redundant if any resource is allowed

omarisgreat commented on StructureDefinition/NHSDigital-Task-Patientcare.StructureDefinition.xml:
```diff
@@ -1,14 +1,14 @@
+  <url value="https://fhir.nhs.uk/StructureDefinition/England-Task-PatientCare" />
   <name value="NHSDigitalTaskPatientCare" />
   <status value="draft" />
   <purpose value="Profile on NHSDigital-Task for NHS App/e-RS PatientCare" />
```
Needs updating

omarisgreat commented on ValueSet/NHSDigital-CareSettingType.xml:
```diff
@@ -1,9 +1,9 @@
+    <title value="England-CareSettingType" />
     <status value="draft" />
     <date value="2020-11-04T00:00:00+00:00" />
     <publisher value="NHS Digital" />
```
Publisher should be updated

omarisgreat commented on ValueSet/NHSDigital-OperationOutcome-Codes.xml:
```diff
@@ -1,11 +1,11 @@
+    <description value="England OperationOutcome Codes"/>
     <compose>
         <include>
             <system value="https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode"/>
```
The CodeSystem NHSD-API-ErrorOrWarningCode needs changing to reference England instead, additionally, All CodeSystems and ValueSets also need a consistent naming convention to match the Profile StructureDefinitions

omarisgreat commented on ValueSet/ValueSet-DM-medicationdispense-code.xml:
```diff
@@ -1,6 +1,6 @@
-    <url value="https://fhir.nhs.uk/ValueSet/NHSDigital-MedicationDispense-Code" />
+    <url value="https://fhir.nhs.uk/ValueSet/England-MedicationDispense-Code" />
     <version value="1.0.1" />
     <name value="DMMedicationDispenseCode" />
```
Should ValueSet names match the convention used within the URL

omarisgreat reviewed (changes requested):
I feel you should probably search/find all occurrences of 'NHSDigital' and 'NHS Digital' in all files in the branch and remove or rename to England if possible. 

Additionally, you should remove references to the base Reference resource in differentials (search for <profile value="https://fhir.nhs.uk/StructureDefinition/NHSDigital-Reference" />).

Finally, naming conventions for Extensions, ValueSets and CodeSystems should also be aligned with the England convention for Profiles to maintain consistency across our assets. e.g. <ResourceType>-<England>-<Programme>-<Name> => Extension-England-EPS-PrescriptionType.

declankieran reviewed (approved):
Adding first approve

declankieran reviewed (approved):
Adding first approve

omarisgreat reviewed (approved):
Approving PR as requested but comments will need to be addressed in further PRs: 

I feel you should probably search/find all occurrences of 'NHSDigital' and 'NHS Digital' in all files in the branch and remove or rename to England if possible.

Additionally, you should remove references to the base Reference resource in differentials.

Finally, naming conventions for Extensions, ValueSets and CodeSystems should also be aligned with the England convention for Profiles to maintain consistency across our assets. e.g. --- => Extension-England-EPS-PrescriptionType.
